lib/puppet/parser/functions/evaluate_version.rb

```diff
@@ -0,0 +1,32 @@
+require_relative 'util/nodejs_functions'
+
+module Puppet::Parser::Functions
+  newfunction(:evaluate_version, :type => :rvalue) do |args|
+    raise(Puppet::ParseError, "evaluate_version(): too few arguments") if args.size < 1
+
+    version = args[0]
+    if version == 'latest'
+      return get_latest_version
+    end
+
+    if version == 'lts'
+      return get_lts_version
+    end
+
+    if version =~ /^(?:(v)?)[0-9]+\.[0-9]+\.[0-9]+/
+      # if the version is matched, but contains no `v` as prefix, it will
+      # be added automatically
+      if version =~ /^[^v](.*)/
+        return 'v' + version
+      end
+
+      return version
+    end
+
+    if version =~ /^(?:(v)?)[0-9]+\.([0-9]+|x)$/
+      return get_version_from_branch version
+    end
+
+    raise Puppet::ParseError, "evaluate_version(): version must be `lts`, `latest` or look like `x.x.x`"
```

Comment:
the exception message is not detailed enough concerning the generic versions. This should be improved.